src/converter/internal/convertercontroller.cpp

```diff
@@ -220,6 +229,9 @@ RetVal<ConverterController::BatchJob> ConverterController::parseBatchJob(const m
         job.in = correctUserInputPath(obj["in"].toString());
         job.out = correctUserInputPath(obj["out"].toString());
 
+        QJsonDocument doc(obj["transpose"].toObject());
+        job.transposeOptions = doc.toJson(QJsonDocument::Compact).toStdString();
```

Comment:
You can also use the proposed new `ConverterUtils` class to avoid converting back and forth between JSON and strings:
- change the interface of `applyTranspose` to take a `TransposeOptions` struct rather than a JSON string
- make `parseTransposeOptions` a public method of `ConverterUtils`
- introduce a second overload of `parseTransposeOptions` that takes `QJsonObject`
- let the original overload (that takes a string) call the new overload internally
- in convertercontroller.cpp and backendapi.cpp,  call the `parseTransposeOptions` overloads as appropriate

Reply:
I have some questions...
- parseTransposeOptions is currently called only by applyTranspose (not within convertercontroller.cpp or backendapi.cpp)
- I don't understand if you mean that we should call applyTranspose with a QJSonObject instead of a std::string to avoid converting back and forth between JSON and strings..

Could you please clarify ?

Reply:
I'll do my best and let you review

Reply:
An alternative: keep `parseTransposeOptions` inside `applyTranspose`, but create two overloads of `applyTranspose`. Perhaps that's simpler and better. The `std::string` overload would call the `QJsonObject` overload to prevent code duplication.

Reply:
Ok, I'll do the second option.

Reply:
Done